Add Advanced Table Resizing Documentation (#2918)
hashicorp/design-system: website/docs/components/table/advanced-table/partials/specifications/resizing-columns.md

Reviewer: [Suggestion] We should probably update the anatomy specifications for the table headers to include the context menu.

Author: I've updated the anatomy for this, and included a bit more detail on the context menu.

## Column resizing

The Advanced Table supports resizing individual columns to display the entire contents of a cell.

> The Ember component supports this via an interactive "resize border" which functions either by clicking and dragging on the horizontal axis with a mouse, or by moving focus to the resize border with a keyboard and using the right and left arrow keys.

Reviewer 2: ```suggestion
This is supported via an interactive "resize border" that functions either by clicking and dragging on the horizontal axis with a mouse or by moving the focus to the resize border with the keyboard and using the right and left arrow keys.
```

The Figma component does not support a specific property to communicate resizing. Instead, we publish a [Resize Border](#jory-to-update-this-link-when-figma-is-published) component individually for use in prototypes or interactive demonstrations. [Templates](#jory-to-update-this-link-when-figma-is-published) are available on the Advanced Table stickersheet to provide a starting point for expressing this interaction.

![The interactive resize border in its active state being dragged with a mouse](/assets/components/table/advanced-table/advanced-table-resize-interaction.png)

### Minimum and maximum width

> To prevent a column from being resized beyond a reasonable amount, the Advanced Table enforces a default minimum and maximum width of `150px` and `800px` respectively, but can be overridden if necessary via the [component API](/components/table/advanced-table?tab=code#advancedtable).

Reviewer 2: ```suggestion
To prevent a column from being resized beyond a reasonable amount, the Advanced Table enforces a default minimum and maximum width of `150px` and `800px`, respectively. These can be overridden via the [component API](/components/table/advanced-table?tab=code#advancedtable), if necessary.
```

> To prevent a column from being resized beyond a reasonable amount, the Advanced Table enforces a default minimum and maximum width of {INSERT MIN HERE} and {INSERT MAX HERE} respectively, but can be overridden if necessary via the [component API](#add-a-link-here).

Author: Yep, I caught that and updated

![An example of a column being resized to the minimum default width](/assets/components/table/advanced-table/advanced-table-resize-min-width.png)

### Content wrapping

> By default, content within the cells of the Advanced Table will wrap according to the browser’s natural reflow. This may result in content breaking to multiple lines and layout shift within the table, especially when `hasResizableColumns` is set to `true` in the Ember component.

Reviewer 2: ```suggestion
By default, content within the cells of the Advanced Table will wrap according to the browser’s natural reflow. This may result in the layout shifting.
```

> How cell content responds to resizing is determined by the implementation. For example, truncation with an ellipsis can be achieved by applying custom CSS (e.g., `text-overflow: ellipsis; white-space: nowrap; overflow: hidden;`) to the relevant element within the table cell. The component provides flexibility to support a range of approaches based on project requirements.

Reviewer 2: ```suggestion
How resizing for the cell content works is determined by the implementation. For example, truncation with an ellipsis can be achieved by applying custom CSS to the relevant element within the table cell, e.g., `text-overflow: ellipsis; white-space: nowrap; overflow: hidden;`. 
```
